src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -467,9 +468,32 @@ public synchronized String using(@NonNull String script, @NonNull Language langu
             // Probably need not add to pendingScripts, since generally that would have happened already in configuring.
             throw new UnapprovedUsageException(hash);
         }
+
+        return script;
+    }
+    /**
+     * Called when a script is about to be used (evaluated).
+     * @param script a possibly unapproved script
+     * @param language the language in which it is written
+     * @param run the run executing the groovy script.
+     * @return {@code script}, for convenience
+     * @throws UnapprovedUsageException in case it has not yet been approved
+     */
+    public synchronized String using(@NonNull String script, @NonNull Language language, @NonNull Run run) throws UnapprovedUsageException {
```

Comment:
How can `run` be `@NonNull`? Wouldn't that require all scripts to necessarily run in the context of a build?

Reply:
Not properly released, but https://github.com/daniel-beck/sandboxed-script-console-plugin demonstrates it's definitely doable.

https://plugins.jenkins.io/scriptler/ has a dependency on Script Security, so there's probably a sandboxed script execution outside builds there as well.

Reply:
Fixed in e69d7cb

Reply:
Thanks for the explanation.

I decided to make the script origin a String and not a Run.

One thing that bugs me a bit is that possible implementations must pass the source as a binding variable instead of a parameter. What do you think: Should it be a String parameter instead?

Reply:
Decided to keep it like this as it is the best solution for now where the endpoint is only (actively) used in combination with builds. It is compatible with other approaches so I think it is fine for now.